case_folding.go

```diff
@@ -3,475 +3,475 @@ package quamina
 // Code generated by code_gen/code_gen - DO NOT EDIT.
 // built from the "C" records in CaseFolding.txt in the Unicode character database
 var caseFoldingPairs = map[rune]rune{
```

Comment:
This is a symptom of a git mistake.  The lint checker changed in GitHub but I hadn't updated my local copy, so all of a sudden there were these CI failures.  So I did PR #348  that addressed just the lint problems.  But I neglected that I also had those in my working copy where I was working with numbits. So they showed as a diff in this PR, even though they are already in the  origin repo. I tried to fix with `git pull` but no luck.  So you are correct that this is unrelated but it isn't really in the PR.  If you have advice how to manipulate with git to avoid this, that would be good.

Reply:
You could do a soft reset, add everything you need selectively and commit then. followed by a force push to overwrite the branch. and do all that on a new local branch to make sure you can still revert back to it if you stumble somewhere. 
But there's a simpler, hacky approach:

git fetch origin/main --prune
git checkout -b tmp origin/main
git checkout numbits -- <list of files you want>
git add <list of files you want>
git commit
git push -f origin tmp:numbits

That's probably nicer to navigate than with revert etc - directly pick the files you want in a new branch instead of wrangling commits.
requires a bit of cleanup because your local numbits branch will be different to the server one and you should get rid of the tmp branch. But I don't want to add the commands to do destructive operations on your local repo.

If you add stuff and you have multiple changes in your repo, add files individually. add -p can also be used to only add regions in a file. And then, take care to not add -a to commit - that will drag everything else back in.

Reply:
Since I don't understand what's really going on in that sequence, unless you think it's really damaging, I think I'd prefer to live with the noisy PR. I'm a git idiot.